native/common/include/jp_javaframe.h

```diff
@@ -65,15 +69,39 @@ class JPJavaFrame
 	 * @throws JPypeException if the jpype cannot
 	 * acquire an env handle to work with jvm.
 	 */
-	JPJavaFrame(JPContext* context, int size = LOCAL_FRAME_DEFAULT);
+	static JPJavaFrame outer(JPContext* context, int size = LOCAL_FRAME_DEFAULT)
+	{
+		return JPJavaFrame(context, NULL, size, true);
+	}
```

Comment:
In order to make sure we can't leak Java objects we need to segregate frames which are inner (controlling scope) and outer (meaning no java objects can survive.   I have switched these to named constructors so that it is clear which is which,

Reply:
That's a very meaningful change. Thanks!